File: Rules/Prepare_Annotation.py
#----------------------------------------------------------------------------- #
rule link_annotation:
        input:
            annotation = ANNOTATION,
        output:
            outfile = os.path.join(PATH_ANNOTATION, 'GTF_Link.gtf')
        params:
            threads   = 1,
            mem       = '16G',
        message:"""
                Running: link_annotation:
                    output: {output.outfile}
            """
        shell:"""
            ln -s {input.annotation} {output.outfile}
        """


#----------------------------------------------------------------------------- #
rule prepare_annotation:
        input:
            gtf_path = rules.link_annotation.output.outfile
        output:
            outfile = os.path.join(PATH_ANNOTATION, 'Processed_Annotation.rds')
        params:
            threads   = 1,
            mem       = '16G',
            scriptdir = SCRIPT_PATH,
            Rscript   = SOFTWARE['Rscript']['executable']
        log:
            log = os.path.join(PATH_LOG, 'prepare_annotation.log')
        message:"""
                Running: prepare_annotation:
                    output: {output.outfile}
            """
        run:
            RunRscript(input, output, params, 'Prepare_Annotation.R')

File: pipelines/chipseq/scripts/Prepare_Annotation.R
# ---------------------------------------------------------------------------- #
options = commandArgs(trailingOnly=TRUE)
source(file.path(options[2],'/scripts/Argument_Parser.R'))
argv = Parse_Arguments('Prepare_Annotation')

# ---------------------------------------------------------------------------- #
Prepare_Annotation = function(
    gtf_path=NULL,
    outfile,
    scriptdir
){

    suppressPackageStartupMessages(library('genomation'))
    suppressPackageStartupMessages(library('GenomicRanges'))
    suppressPackageStartupMessages(library('data.table'))
    source(file.path(scriptdir,'Functions_Helper.R'), local=TRUE)

    message('Annotation...')
    if(is.null(gtf_path))
        stop('Annotation is not specified')

    annotation = ReadGTFAnnotation(gtf_path)

    annotation$genomic_annotation = GTFGetAnnotation(annotation$gtf)

    # --------------------------------------------------------------- #
    message('RDS ...')
    saveRDS(annotation, outfile)
}


# ---------------------------------------------------------------------------- #
# function call
Prepare_Annotation(
    gtf_path  = argv$input[['gtf_path']],
    outfile   = argv$output[['outfile']],
    scriptdir = argv$params[['scriptdir']]
)
